File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {

    List<List<Integer>> res;

    public List<List<Integer>> combinationSum(int[] nums, int target) {
        res = new ArrayList<>();
        List<Integer> subset = new ArrayList<>();
        dfs(nums, target, 0, subset, 0);
        return res;
    }

    public void dfs(int[] nums, int target, int rs, 
    List<Integer> subset, int i) {
        if (rs == target) {
            res.add(new ArrayList<>(subset));
            return;
        } else if (rs > target) {
            return;
        }
        if (i >= nums.length) return;

        // for (int j = i; j < nums.length; j++) {
        subset.add(nums[i]);
        dfs(nums, target, rs + nums[i], subset, i);
        subset.remove(subset.size() - 1);
        dfs(nums, target, rs, subset, i+1);
        // }
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result = set()

        for i in range(len(nums)-2):
            for j in range(i + 1, len(nums) - 1):
                for k in range(j + 1, len(nums)):
                    if nums[i] + nums[j] + nums[k] == 0:
                        result.add((nums[i], nums[j], nums[k]))
        return [list(res) for res in list(result)]